upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -234,6 +239,34 @@ func (c *ApplyClusterCmd) Run(ctx context.Context) error {
 
 	cluster := c.Cluster
 
+	configBase, err := vfs.Context.BuildVfsPath(cluster.Spec.ConfigBase)
+	if err != nil {
+		return fmt.Errorf("error parsing config base %q: %v", cluster.Spec.ConfigBase, err)
+	}
+
+	if !c.AllowKopsDowngrade {
+		kopsVersionApplied, err := configBase.Join(registry.PathKopsVersionApplied).ReadFile()
+		if err == nil {
+			version, err := semver.Parse(strings.TrimSpace(string(kopsVersionApplied)))
+			if err != nil {
+				return fmt.Errorf("error parsing last kops version applied: %v", err)
+			}
+			if version.GT(semver.MustParse(kopsbase.Version)) {
+				fmt.Printf("\n")
+				fmt.Printf("%s\n", starline)
+				fmt.Printf("\n")
+				fmt.Printf("The cluster was last updated by kops version %s\n", kopsVersionApplied)
+				fmt.Printf("To permit updating by the older version %s, run with the --allow-kops-downgrade flag\n", kopsbase.Version)
+				fmt.Printf("\n")
+				fmt.Printf("%s\n", starline)
+				fmt.Printf("\n")
+				return fmt.Errorf("kops version older than that last used to apply cluster")
+			}
+		} else if err != os.ErrNotExist {
+			return fmt.Errorf("error reading last kops version applied: %v", err)
```

Comment:
This file will not exist before applying this fix the fist time. Maybe should just ignore it?
```suggestion
			klog.Infof("Last used kops version not found; will assume older version")
```

Reply:
The `!=` comparison causes it to ignore `os.ErrNotExist`

Reply:
Ah, missed that.